tests/test_backends.py

```diff
@@ -0,0 +1,63 @@
+from typing import Tuple, cast
+
+import os
+import pulser as pl
+import pytest
+import conftest
+import torch_geometric.data as pyg_data
+import torch_geometric.datasets as pyg_dataset
+from qek.backends import CompilationError, QutipBackend, BaseBackend
+import qek.data.graphs as qek_graphs
+
+if os.name == "posix":
+    # As of this writing, emu-mps only works under Unix.
+    from qek.backends import EmuMPSBackend
+
+
+@pytest.mark.asyncio
+async def test_async_emulators() -> None:
+    """
+    Test that backends based on emulators can execute without exploding (async).
+    """
+    conftest.preload_dataset()
+
+    # Load dataset
+    original_ptcfm_data = [
+        cast(pyg_data.Data, d) for d in pyg_dataset.TUDataset(root="dataset", name="PTC_FM")
+    ]
+
+    compiled: list[Tuple[qek_graphs.BaseGraph, pl.Register, pl.Pulse]] = []
+    for i, data in enumerate(original_ptcfm_data):
+        print("loading graph %s" % (i,))
+        graph = qek_graphs.PTCFMGraph(data=data, device=pl.AnalogDevice, id=i)
+        try:
+            register = graph.compile_register()
```

Comment:
Out of curiosity, is it a requirement that the compilation for the register is let to be called manually ? What I mean is wouldn't that be easier if that were triggered at graph instantiation and then we can simply do `register = graph.register` ?

Reply:
We could certainly do that. Initially, I wanted to reserve the option of setting up the geometry manually after creating the graph, but we don't exploit this option at the moment.

Reply:
Filed as #74.